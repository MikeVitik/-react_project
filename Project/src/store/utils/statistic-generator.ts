import { TASK_TIME, TOTAL_MILLISECONDS_IN_DAY } from "../const";
import { StatisticItem } from "../slicies/statistic";

export function generateStatistics(
  currentDate = new Date(Date.now() - TOTAL_MILLISECONDS_IN_DAY),
  seed = 1
): StatisticItem[] {
  function random(max: number) {
    var x = Math.sin(seed++) * 10000;
    return Math.floor(max * (x - Math.floor(x)));
  }

  const result: StatisticItem[] = [];
  let id = -1;
  for (let day = 0; day < 7 * 3; day++) {
    for (let attemps = 0; attemps < 5; attemps++) {
      const isPomodoroComplete = random(3) <= 1;
      const startDateString = currentDate.toUTCString();
      if (isPomodoroComplete) {
        result.push({
          taskId: id,
          startDateString,
          type: "work",
          isPomodoroComplete: true,
          workTime: TASK_TIME,
        } satisfies StatisticItem);
      } else {
        const workTime = random(TASK_TIME);
        result.push(
          {
            taskId: id,
            startDateString,
            type: "work",
            workTime,
          } satisfies StatisticItem,
          {
            taskId: id,
            startDateString,
            type: "pause",
            pauseTime: TASK_TIME - workTime,
          } satisfies StatisticItem
        );
      }
    }
    currentDate = new Date(currentDate.getTime() - TOTAL_MILLISECONDS_IN_DAY);
    id = id - 1;
  }
  return result.reverse();
}
