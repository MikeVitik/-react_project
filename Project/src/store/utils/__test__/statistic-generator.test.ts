import { generateStatistics } from "../statistic-generator";

describe("generateStatistic", () => {
  it("should generate statistic for 2 weeks", () => {
    const items = generateStatistics(new Date(2024, 1, 20));
    expect(items.filter(({ type }) => type === "pause").length).toBeGreaterThan(
      0
    );
    expect(items.filter(({ type }) => type === "work").length).toBeGreaterThan(
      0
    );
    expect(
      items.filter(
        ({ completedPomodoro, workTime }) => completedPomodoro === workTime
      ).length
    ).toBeGreaterThan(0);
    expect(items).toMatchSnapshot();
  });
});
