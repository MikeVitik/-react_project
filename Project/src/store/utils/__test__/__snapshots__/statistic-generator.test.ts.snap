// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateStatistic should generate statistic for 2 weeks 1`] = `
Array [
  Object {
    "pauseTime": 1425108,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "work",
    "workTime": 74892,
  },
  Object {
    "pauseTime": 905109,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "work",
    "workTime": 594891,
  },
  Object {
    "pauseTime": 783893,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "work",
    "workTime": 716107,
  },
  Object {
    "pauseTime": 498101,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "work",
    "workTime": 1001899,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "taskId": -21,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 513657,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "work",
    "workTime": 986343,
  },
  Object {
    "pauseTime": 50808,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "work",
    "workTime": 1449192,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1048992,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "taskId": -20,
    "type": "work",
    "workTime": 451008,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1304804,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "pause",
  },
  Object {
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "work",
    "workTime": 195196,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1332237,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "pause",
  },
  Object {
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "taskId": -19,
    "type": "work",
    "workTime": 167763,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 969990,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "pause",
  },
  Object {
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "work",
    "workTime": 530010,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "taskId": -18,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1222419,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "pause",
  },
  Object {
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "taskId": -17,
    "type": "work",
    "workTime": 277581,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 336048,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "work",
    "workTime": 1163952,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "taskId": -16,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 984617,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "pause",
  },
  Object {
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "work",
    "workTime": 515383,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1383915,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "pause",
  },
  Object {
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "work",
    "workTime": 116085,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "taskId": -15,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 991045,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "work",
    "workTime": 508955,
  },
  Object {
    "pauseTime": 50046,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "work",
    "workTime": 1449954,
  },
  Object {
    "pauseTime": 25459,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "work",
    "workTime": 1474541,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "taskId": -14,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1134300,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "work",
    "workTime": 365700,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "taskId": -13,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 190031,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "pause",
  },
  Object {
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "work",
    "workTime": 1309969,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "taskId": -12,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 680202,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "pause",
  },
  Object {
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "work",
    "workTime": 819798,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 415211,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "pause",
  },
  Object {
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "taskId": -11,
    "type": "work",
    "workTime": 1084789,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "taskId": -10,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "taskId": -10,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "taskId": -10,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "taskId": -10,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "taskId": -10,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 266551,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "work",
    "workTime": 1233449,
  },
  Object {
    "pauseTime": 929893,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "work",
    "workTime": 570107,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "taskId": -9,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "taskId": -8,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "taskId": -8,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "taskId": -8,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "taskId": -8,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "taskId": -8,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 403159,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "pause",
  },
  Object {
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "work",
    "workTime": 1096841,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "taskId": -7,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 323219,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "work",
    "workTime": 1176781,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 69206,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "pause",
  },
  Object {
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "work",
    "workTime": 1430794,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "taskId": -6,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1240043,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "pause",
  },
  Object {
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "work",
    "workTime": 259957,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 564680,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "pause",
  },
  Object {
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "taskId": -5,
    "type": "work",
    "workTime": 935320,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1413176,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "pause",
  },
  Object {
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "work",
    "workTime": 86824,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1175431,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "pause",
  },
  Object {
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "taskId": -4,
    "type": "work",
    "workTime": 324569,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 821240,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "pause",
  },
  Object {
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "work",
    "workTime": 678760,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "taskId": -3,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 49750,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "work",
    "workTime": 1450250,
  },
  Object {
    "pauseTime": 1389665,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "work",
    "workTime": 110335,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 1353099,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "pause",
  },
  Object {
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "work",
    "workTime": 146901,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "taskId": -2,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 201020,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "pause",
  },
  Object {
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "work",
    "workTime": 1298980,
  },
  Object {
    "pauseTime": 364120,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "pause",
  },
  Object {
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "work",
    "workTime": 1135880,
  },
  Object {
    "isPomodoroComplete": true,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "work",
    "workTime": 1500000,
  },
  Object {
    "pauseTime": 38598,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "pause",
  },
  Object {
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "taskId": -1,
    "type": "work",
    "workTime": 1461402,
  },
]
`;
